command/init.go

```diff
@@ -630,13 +631,38 @@ func (c *InitCommand) getProviders(config *configs.Config, state *states.State,
 				default: // version is less than closestAvailable
 					diags = diags.Append(tfdiags.Sourceless(
 						tfdiags.Error,
-						"Incompatible provider version",
+						summaryIncompatible,
 						fmt.Sprintf(providerProtocolTooOld, provider.ForDisplay(),
 							version, tfversion.String(), closestAvailable, closestAvailable,
 							getproviders.VersionConstraintsString(reqs[provider]),
 						),
 					))
 				}
+			case getproviders.ErrPlatformNotSupported:
+				switch {
+				case err.MirrorURL != nil:
+					// If we're installing from a mirror then it may just be
+					// the mirror lacking the package, rather than it being
+					// unavailable from upstream.
+					diags = diags.Append(tfdiags.Sourceless(
+						tfdiags.Error,
+						summaryIncompatible,
+						fmt.Sprintf(
+							"Your chosen provider mirror at %s does not have a %s v%s plugin available for your current platform, %s.\n\nProvider releases are separate from Terraform CLI releases, so this provider might not support your current platform. Alternatively, the mirror itself might have only a subset of the plugin packages available in the origin registry, at %s.",
```

Comment:
I believe we mostly use the phrases "provider" or "provider plugin", instead of just "plugin", in provider installer messages, so the use of "plugin" here jumped out at me. Do you have any thoughts here? I think mostly we use "plugin" when it's referring to something related to both providers and provisioners, like the plugin lock file. 

This is not a merge-blocking question!

Reply:
Hmm... I had assumed there was enough "provider-y" context here that it would be redundant to include that modifier again in what is already a pretty-verbose message. I'm happy to change it if you disagree, though.

On re-read with fresh eyes today I actually find myself thinking that I should've used the word "package" instead of "plugin" here, because that is the word we've typically used for the specific artifact we download, as opposed to the set of functionality it implies. :thinking:

Reply:
that works too!